### PR TITLE
HPCC-15322 Allow Dali backup queue limits to be configured

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith jake.smith@lexisnexis.com

### CODE REVIEW

**Comment by jakesmith** [comment]
@mckellyln - please review

File: dali/base/dasds.cpp [hunk 1]
```diff
@@ -1040,6 +1040,12 @@ class CBackupHandler : public CInterface, implements IThreaded
     CTimeMon warningTime;
     unsigned recentTimeThrottled;
     unsigned lastNumWarnItems;
+    IPropertyTree &config;
+
+    const unsigned defaultFreeQueueLimit = 50;
```
Comment (mckellyln) [inline dali/base/dasds.cpp]: Why 50 and not 10 to match previous ?

Comment (jakesmith) [inline dali/base/dasds.cpp]: 10 was too conservative.

File: initfiles/componentfiles/configxml/dali.xsd [hunk 2]
```diff
@@ -303,6 +303,27 @@
...
+      </xs:annotation>
+    </xs:attribute>
+    <xs:attribute name="backupSoftQueueLimitDelay" type="xs:nonNegativeInteger" use="optional" default="200">
+      <xs:annotation>
+        <xs:appinfo>
+          <tooltip>The maximum delay to introduce when the backupSoftQueueLimit has been introduced</tooltip>
+        </xs:appinfo>
+      </xs:annotation>
+    </xs:attribute>
   </xs:attributeGroup>
```
Comment (mckellyln) [inline initfiles/componentfiles/configxml/dali.xsd]: Is backupFreeQueueLimit missing ?

Comment (jakesmith) [inline initfiles/componentfiles/configxml/dali.xsd]: I deliberately left out. I thought too 'internal'

File: initfiles/componentfiles/configxml/dali.xsl [hunk 3]
```diff
@@ -123,7 +123,7 @@
             <xsl:with-param name="val" select="@recoverFromIncErrors"/>
           </xsl:call-template>
         </xsl:attribute>
-        <xsl:copy-of select="@snmpErrorMsgLevel | @msgLevel | @lightweightCoalesce | @keepStores"/>
+        <xsl:copy-of select="@snmpErrorMsgLevel | @msgLevel | @lightweightCoalesce | @keepStores | @backupLargeWarningThreshold | @backupSoftQueueLimit | @backupSoftQueueLimitDelay"/>
         <xsl:attribute name="lCIdlePeriod">
```
Comment (mckellyln) [inline initfiles/componentfiles/configxml/dali.xsl]: Same as above, why not backupFreeQueueLimit ?

**Comment by mckellyln** [comment]
@jakesmith a few comments inline.

**Comment by jakesmith** [comment]
Deliberately left 'backupFreeQueueLimit' from exposed config, as I think it's too meaningless / obscure to be exposed.
Changed (previously hard wired) default freeQueueLimit from 10 to 50. It's a bit arbitrary, but gives a few more in pool to reuse.

@richardkchapman , good to merge I think.

File: dali/base/dasds.cpp [hunk 4]
```diff
@@ -5720,7 +5726,7 @@ IStoreHelper *createStoreHelper(const char *storeName, const char *location, con
 #endif
 
 CCovenSDSManager::CCovenSDSManager(ICoven &_coven, IPropertyTree &_config, const char *_dataPath) 
-    : coven(_coven), config(_config), server(*this), dataPath(_dataPath)
+    : coven(_coven), config(_config), server(*this), dataPath(_dataPath), backupHandler(config)
```
Comment (mckellyln) [inline dali/base/dasds.cpp]: backupHandler(_config) ??

Comment (jakesmith) [inline dali/base/dasds.cpp]: Yep, I agree, no guarantee about initialization order.
Have pushed change and squashed.

**Comment by mckellyln** [comment]
Looks good to merge.

**Pull request merged**